lib/iris/_data_manager.py

```diff
@@ -0,0 +1,471 @@
+                emsg = ('Can only cast lazy data to an integer or boolean '
+                        'dtype, got {!r}.')
+                raise ValueError(emsg.format(realised_dtype))
+            self._realised_dtype = realised_dtype
+
+            # Check the manager contract, as the managed dtype has changed.
+            self._assert_axioms()
+
+    @property
+    def core_data(self):
+        """
+        If real data is being managed, then return the :class:`~numpy.ndarray`
+        or :class:`numpy.ma.core.MaskedArray`. Otherwise, return the lazy
+        :class:`~dask.array.core.Array`.
+
+        Returns:
+            The real or lazy data.
+
+        """
+        if self.has_lazy_data():
+            result = self._lazy_array
+        else:
+            result = self._real_array
+
+        return result
+
+    @property
+    def data(self):
+        """
+        Returns the real data. Any lazy data being managed will be realised.
+
+        Returns:
+            :class:`~numpy.ndarray` or :class:`numpy.ma.core.MaskedArray`.
+
+        """
+        if self.has_lazy_data():
+            try:
+                # Realise the lazy data.
+                result = as_concrete_data(self._lazy_array,
+                                          nans_replacement=ma.masked,
+                                          result_dtype=self.dtype)
+                # Assign the realised result.
+                self._real_array = result
+                # Reset the lazy data and the realised dtype.
+                self._lazy_array = None
+                self._realised_dtype = None
+            except MemoryError:
+                emsg = ('Failed to realise the lazy data as there was not '
+                        'enough memory available.\n'
+                        'The data shape would have been {!r} with {!r}.\n '
+                        'Consider freeing up variables or indexing the data '
+                        'before trying again.')
+                raise MemoryError(emsg.format(self.shape, self.dtype))
+
+            # Check the manager contract, as the managed data has changed.
+            self._assert_axioms()
+
+        return self._real_array
+
+    @data.setter
+    def data(self, data):
+        """
+        Replaces the currently managed data with the specified data, which must
+        be of an equivalent shape.
+
+        Note that, the only shape promotion permitted is for 0-dimensional
+        scalar data to be replaced with a single item 1-dimensional data.
+
+        Args:
+
+        * data:
+            The :class:`~numpy.ndarray` or :class:`~numpy.ma.core.MaskedArray`
+            real data, or :class:`~dask.array.core.Array` lazy data to be
+            managed.
+
+        """
+        # Ensure we have numpy-like data.
+        if not (hasattr(data, 'shape') and hasattr(data, 'dtype')):
+            data = np.asanyarray(data)
+
+        # Determine whether the class instance has been created,
```

Comment:
It feels like there should be a better way to check for instantiation than this...